src/magicbox/SuggestionsManager.ts

```diff
@@ -379,23 +379,30 @@ export class SuggestionsManager {
 
   private updateAreaSelectedIfDefined(element: HTMLElement, value: string): void {
     if ($$(element).getAttribute('aria-selected')) {
-      $$(this.inputManager.input).setAttribute('aria-activedescendant', element.id);
+      this.inputManager.activeDescendant = element;
       $$(element).setAttribute('aria-selected', value);
     }
   }
 
-  private addAccessibilityPropertiesForCombobox() {
-    const combobox = $$(this.magicBoxContainer);
-    const input = $$(this.inputManager.input);
+  private addAccessibilityProperties() {
+    this.addAccessibilityPropertiesForMagicBox();
+    this.addAccessibilityPropertiesForInput();
+  }
+
+  private addAccessibilityPropertiesForMagicBox() {
+    const magicBox = $$(this.magicBoxContainer);
 
-    combobox.setAttribute('aria-expanded', 'false');
-    combobox.setAttribute('role', 'combobox');
-    combobox.setAttribute('aria-owns', 'coveo-magicbox-suggestions');
-    combobox.setAttribute('aria-haspopup', 'listbox');
+    magicBox.setAttribute('role', 'search');
+    magicBox.setAttribute('aria-haspopup', 'listbox');
+  }
+
+  private addAccessibilityPropertiesForInput() {
+    const input = $$(this.inputManager.input);
 
-    input.el.removeAttribute('aria-activedescendant');
+    this.inputManager.activeDescendant = null;
+    this.inputManager.expanded = false;
+    input.setAttribute('aria-owns', 'coveo-magicbox-suggestions');
     input.setAttribute('aria-controls', 'coveo-magicbox-suggestions');
-    input.setAttribute('aria-autocomplete', 'list');
   }
 
```

Comment:
It would be good to add unit tests asserting that the right accessibility attributes are set for the important cases,
e.g.

Input when suggestions are not expanded.
Input with suggestions expanded and i) suggestions are available, ii) no suggestions are available.

Reply:
Good point, I'll add unit tests to make sure all those properties are set correctly.